services/explorer-indexer/src/modules/node/historicalSyncer.service.ts

```diff
@@ -163,114 +174,227 @@ export default class HistoricalSyncService {
+                number
+                hash
+                stateRoot
+                gasLimit
+                gasUsed
+                timestamp
+                transactionCount
+                parent {
+                  hash
+                }
+                difficulty
+                totalDifficulty
+                miner {
+                  address
+                }
+                transactionsRoot
+                extraData
+                logsBloom
+                mixHash
+                receiptsRoot
+                ommerHash
+                stateRoot
+                transactions {
+                  block {
+                    hash
+                    number
+                  }
+                  hash
+                  nonce
+                  from {
+                    address
+                  }
+                  to {
+                    address
+                  }
+                  gas
+                  gasUsed
+                  cumulativeGasUsed
+                  gasPrice
+                  inputData
+                  r
+                  v
+                  s
+                  index
+                  value
+                }
+              }
+            }
+          `,
+          {
+            from: this.syncState.number,
+            to: this.chainState.number,
+          }
+        )
+      )?.blocks;
+      console.log(blocksWithTransactions);
+
+      const iBlocks = blocksWithTransactions.map((block) =>
+        this.pbftService.pbftGQLToIPBFT(block)
       );
+      for (const block of iBlocks) {
+        const started = new Date();
+        let blockReward = new BigInteger('0', 10);
 
-      const block = this.pbftService.pbftRpcToIPBFT(blockRpc);
-      this.logger.log(
-        `Getting ${block.transactions.length} transactions for block ${
-          this.syncState.number + 1
-        }...`
-      );
-      const blockTxs: ITransaction[] = [];
-      const fetchTxStart = new Date();
-      for (let i = 0; i < block.transactions.length; i++) {
-        this.logger.debug(
-          `Getting transaction #${i + 1} out of #${
-            block.transactions.length
-          } - ${block.transactions[i]?.hash}`
+        block.transactions?.map(async (tx) =>
+          this.txService.populateTransactionWithPBFT(tx, block)
         );
-
-        if (block.transactions[i]) {
-          const fetchStart = new Date();
-          const tx = await this.rpcConnector.getTransactionByHash(
-            block.transactions[i].hash
-          );
-          const fetchEnd = new Date();
-          console.log(
-            `Fetched tx ${block.transactions[i].hash} in ${
-              fetchEnd.getTime() - fetchStart.getTime()
-            }ms.`
-          );
-          blockTxs.push(
-            this.txService.populateTransactionWithPBFT(
-              this.txService.txRpcToITransaction(tx),
-              block
-            )
-          );
+        block.transactions?.forEach((tx: ITransaction) => {
```

Comment:
I would create a new transactions array (`formmatedTransactions`) and use .map instead of forEach for performance purposes.
Then in the `blockToSave.transactions = formmatedTransactions;`